x-pack/plugin/inference/qa/inference-service-tests/src/javaRestTest/java/org/elasticsearch/xpack/inference/MockElasticInferenceServiceAuthorizationServer.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.core.Strings;
+import org.elasticsearch.test.http.MockResponse;
+import org.elasticsearch.test.http.MockWebServer;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+import static org.elasticsearch.core.Strings.format;
+
+public class MockElasticInferenceServiceAuthorizationServer implements TestRule {
+
+    private static final Logger logger = LogManager.getLogger(MockElasticInferenceServiceAuthorizationServer.class);
+    private final MockWebServer webServer = new MockWebServer();
+
+    public static MockElasticInferenceServiceAuthorizationServer enabledWithSparseAndChatCompletion() {
+        var server = new MockElasticInferenceServiceAuthorizationServer();
+
+        String responseJson = """
+            {
+                "models": [
+                    {
+                      "model_name": "model-a",
+                      "task_types": ["embed/text/sparse", "chat"]
+                    }
+                ]
+            }
+            """;
+
+        server.webServer.enqueue(new MockResponse().setResponseCode(200).setBody(responseJson));
+        return server;
+    }
+
+    public static MockElasticInferenceServiceAuthorizationServer disabled() {
+        var server = new MockElasticInferenceServiceAuthorizationServer();
+
+        String responseJson = """
+            {
+                "models": []
+            }
+            """;
+
+        server.webServer.enqueue(new MockResponse().setResponseCode(200).setBody(responseJson));
+        return server;
+    }
+
+    public String getUrl() {
+        return format("http://%s:%s", webServer.getHostName(), webServer.getPort());
+    }
+
+    @Override
+    public Statement apply(Statement statement, Description description) {
+        return new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                try {
+                    logger.info("Staring mock EIS gateway");
```

Comment:
```suggestion
                    logger.info("Starting mock EIS gateway");
```

Reply:
![Alt Text](https://media4.giphy.com/media/v1.Y2lkPTc5MGI3NjExamV2ZmNiZG4zcm9rNW13NjVnbHVvZXM0eHhuaDd5aTBiYTU2ZjFndyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/aXUU30cDBa9tVQz37V/giphy.gif)

Staring mock gateway is way more interesting though haha